.github/workflows/ci-pipeline.yml

```diff
@@ -0,0 +1,78 @@
+name: CI Pipeline for DoDo
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+  push:
+    branches:
+      - develop
+
+jobs:
+  test_and_build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Set up Docker Compose
+      run: |
+        docker compose version
+
+    - name: Start Services
+      run: |
+        docker compose up -d --build
+      env:
+        POSTGRES_DB: dodo
+        POSTGRES_USER: dodo_user
+        POSTGRES_PASSWORD: dodo_password
+
+    - name: Wait for PostgreSQL to be ready
+      run: |
+        for i in {1..10}; do
+          docker compose exec db pg_isready -U dodo_user -d dodo && break
+          sleep 5
+        done
```

Comment:
There's already an healthcheck in docker compose so I think in most cases this might not be needed and is an extra step but since it just completes in 0s I guess we can live with this.

Reply:
The Docker Compose healthcheck should already ensure PostgreSQL readiness, making this step redundant. However, this manual check can serve as an extra safeguard in edge cases.